Feature: format phone number

aggarwalRuchir opened the pull request:
Feature: Format phone number

This feature will view the phone numbers in the national phone convention. The convention used is for Singapore but many other countries follow the same convention.
The standard is to view phone numbers as _xxxx-xxxx_.

For example: 98765432 will be viewed as 9876-5432

Commit:
Changes for v1.2

Commit:
Changes between v1.2 and v1.3

Commit:
Phone.java: add method formatPhone
Phone.java: add method generateFormattedPhone
Phone.java: add method isHyphenNeeded
Phone.java: add call to method formatPhone

Commit:
MainApp.java: add version for v1.3
Phone.java: Add newline

Commit:
PhoneTest.java: add import for assertEquals
PhoneTest.java: add method isPhoneFormmatingCorrect

wshijing reviewed (approved):
Can add more validation for different country codes and phone numbers in V1.4 but for a simple separation of numbers should be fine for now:) Good job!!

wshijing commented:
You might want to look at why the travis test failed too:(

Commit:
PhoneTest.java: remove space

Commit:
PhoneTest.java: add 3 more tests for different countries

Commit:
DeveloperGuide.adoc: add details about updated Phone attribute

Commit:
Phone.java: assign value trimmedPhone to this.value in constructor
Phone.java: return string value from formatPhone in toString method

Commit:
GuiTestAssert.java: compare expected Phone to formOriginalPhone in assertCardDisplaysPerson
GuiTestAssert.java: add formOriginalPhone method

Commit:
Phone.java: add newline
GuiTestAssert.java: add whitespace after if condition
GuiTestAssert.java: remove whitespace before ‘)’ condition

Commit:
ListCommandActivityDiagram.png: add activity diagram for list command
ListCommandSequenceDiagram.png: add sequence diagram for list command

Commit:
DeveloperGuide.adoc: add list mechanism

wshijing committed:
Merge branch 'master' into feature_formatPhoneNumber

wshijing committed:
Merge branch 'master' into feature_formatPhoneNumber
